pkgs/development/python-modules/keyring-pass/default.nix

```diff
@@ -0,0 +1,83 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, gnupg
+, keyring
+, pass
+, poetry-core
+, pythonOlder
+}:
+let
+  pname = "keyring-pass";
+  version = "0.8.1";
+  module_name = builtins.replaceStrings [ "-" ] [ "_" ] pname;
+in
+
+buildPythonPackage {
+  inherit pname version;
+  disabled = pythonOlder "3.6";
+  format = "pyproject";
+
+  src = fetchPypi {
+    # results in 404 without this replacement
+    pname = module_name;
+    inherit version;
+    hash = "sha256-SrZBiHehMIKU20R+piF2OEnS/4epf4D7Gz09Id6+ZO8=";
+  };
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  propagatedBuildInputs = [
+    pass
+  ];
+
+  nativeCheckInputs = [
+    keyring
+    gnupg
+    pass
+  ];
+
+  doCheck = true;
+
+  preCheck = ''
+    export PASSWORD_STORE_DIR="$TMPDIR/.password-store"
+    export GNUPGHOME="$TMPDIR/.gnupg"
+
+    # generate temporary GPG identity
+    cat <<EOF | gpg --gen-key --batch /dev/stdin
+    %no-protection
+    %transient-key
+    Key-Type: 1
+    Key-Length: 1024
+    Subkey-Type: 1
+    Subkey-Length: 1024
+    Name-Real: test
+    Name-Email: test@example.com
+    Expire-Date: 1
+    EOF
+    # initialize password store
+    pass init test@example.com
+  '';
+
+  checkPhase = ''
+    runHook preCheck
+
+    # added `__init__` due to: No module named keyring_pass.__main__; 'keyring_pass' is a package and cannot be directly executed
```

Comment:
Does it help if we `rm -r keyring_pass` first and add `$out/...` to `$PYTHONPATH`?

Reply:
while doing so I noticed `$out` is already part of PYTHONPATH